core/domain/classifier_services.py

```diff
@@ -417,12 +445,24 @@ def _add_examples(self, examples, iterations):
     def add_examples_for_training(self, training_examples):
         """Adds examples to the classifier with _training_iterations number of
         iterations.
+
+        Args:
+            training_examples: A list of docs.
+
+        Returns:
+            A xrange object. An iterator over the ids of the given docs.
         """
         return self._add_examples(training_examples, self._training_iterations)
 
     def add_examples_for_predicting(self, prediction_examples):
         """Adds examples to the classifier with _prediction_iterations number
         of iterations.
+
+        Args:
+            prediction_examples: A list of docs.
+
+        Returns:
+            A xrange object, the prediction report (iterator of doc ids).
```

Comment:
I couldn't get this one either. Both **add_examples_for_training** and **add_examples_for_predicting** use the same internal method **_add_examples**, with the diffrence been  for **predicting** the caller does not provide any labels and it use's all the registred labels in the classifier as labels instead and calls **_add_examples**. 
from what i understood **_add_examples** updates the classifier internal state with the given data and runs gibbs sampling for a certain number of iterations , then returns an iterator of the ids of the added docs, and since the docs ids are incremental it just returns ```xrange(number_of_docs_before, current_number_of_docs)```